server/cmd/mmctl/commands/plugin_e2e_test.go

```diff
@@ -379,3 +379,135 @@ func (s *MmctlE2ETestSuite) TestPluginDeleteCmd() {
+		s.Require().Len(plugins.Active, 0)
+		s.Require().Len(plugins.Inactive, 1)
+		s.Require().Len(printer.GetLines(), 1)
+		s.Require().Equal(printer.GetLines()[0], "Disabled plugin: "+pluginID)
+	})
+
+	s.Run("error for disable plugin due insufficient permissions", func() {
+		printer.Clean()
+
+		appErr := s.th.App.EnablePlugin(pluginID)
+		s.Require().Nil(appErr)
+
+		plugins, appErr := s.th.App.GetPlugins()
+		s.Require().Nil(appErr)
+		s.Require().Len(plugins.Active, 1)
+		s.Require().Len(plugins.Inactive, 0)
+
+		cmd := &cobra.Command{}
+		_ = pluginDisableCmdF(s.th.Client, cmd, []string{pluginID})
+		s.Require().Len(printer.GetLines(), 0)
+		s.Require().Len(printer.GetErrorLines(), 1)
+		s.Require().Equal(printer.GetErrorLines()[0], "Unable to disable plugin: "+pluginID+". Error: You do not have the appropriate permissions.")
+
+		plugins, appErr = s.th.App.GetPlugins()
+		s.Require().Nil(appErr)
+		s.Require().Len(plugins.Active, 1)
+		s.Require().Len(plugins.Inactive, 0)
+
+		appErr = s.th.App.DisablePlugin(pluginID)
+		s.Require().Nil(appErr)
+	})
+
+	s.RunForSystemAdminAndLocal("error for disabling non existent plugin", func(c client.Client) {
+		printer.Clean()
+
+		plugins, appErr := s.th.App.GetPlugins()
+		s.Require().Nil(appErr)
+		s.Require().Len(plugins.Active, 0)
+		s.Require().Len(plugins.Inactive, 1)
+
+		cmd := &cobra.Command{}
+		_ = pluginDisableCmdF(c, cmd, []string{nonExistentPluginID})
+		s.Require().Len(printer.GetLines(), 0)
+		s.Require().Len(printer.GetErrorLines(), 1)
+		s.Require().Equal(printer.GetErrorLines()[0], "Unable to disable plugin: "+nonExistentPluginID+". Error: Plugin is not installed.")
+
+		plugins, appErr = s.th.App.GetPlugins()
+		s.Require().Nil(appErr)
+		s.Require().Len(plugins.Active, 0)
+		s.Require().Len(plugins.Inactive, 1)
+	})
+
+	s.RunForAllClients("error when plugins are disabled", func(c client.Client) {
+		printer.Clean()
+
+		appErr := s.th.App.EnablePlugin(pluginID)
+		s.Require().Nil(appErr)
+		plugins, appErr := s.th.App.GetPlugins()
+		s.Require().Nil(appErr)
+		s.Require().Len(plugins.Active, 1)
+		s.Require().Len(plugins.Inactive, 0)
+
+		s.th.App.UpdateConfig(func(cfg *model.Config) {
+			*cfg.PluginSettings.Enable = false
+		})
+
+		cmd := &cobra.Command{}
+		_ = pluginDisableCmdF(c, cmd, []string{pluginID})
+		s.Require().Len(printer.GetLines(), 0)
+		s.Require().Len(printer.GetErrorLines(), 1)
+		s.Require().Equal(printer.GetErrorLines()[0], "Unable to disable plugin: "+pluginID+". Error: Plugins have been disabled. Please check your logs for details.")
+
+		s.th.App.UpdateConfig(func(cfg *model.Config) {
+			*cfg.PluginSettings.Enable = true
+		})
+		plugins, appErr = s.th.App.GetPlugins()
+		s.Require().Nil(appErr)
+		s.Require().Len(plugins.Active, 1)
+		s.Require().Len(plugins.Inactive, 0)
+	})
+}
+
+func removePlugin(s *MmctlE2ETestSuite, pluginID string) {
+	s.th.App.UpdateConfig(func(cfg *model.Config) {
+		*cfg.PluginSettings.Enable = true
+	})
```

Comment:
It's a bit odd that a utility function changes settings. Should the caller do that?